opengrok-indexer/src/main/java/org/opengrok/indexer/history/HistoryGuru.java

```diff
@@ -955,26 +957,19 @@ private void createHistoryCache(Repository repository, String sinceRevision) {
             Statistics elapsed = new Statistics();
 
             LOGGER.log(Level.INFO, "Creating history cache for {0}", repository);
-
-            try {
-                repository.createCache(historyCache, sinceRevision);
-            } catch (Exception e) {
-                LOGGER.log(Level.WARNING,
-                        String.format("An error occurred while creating cache for %s", repository), e);
-            }
-
+            repository.createCache(historyCache, sinceRevision);
             elapsed.report(LOGGER, String.format("Done history cache for %s", repository));
         } else {
             LOGGER.log(Level.WARNING,
                     "Skipping creation of history cache for {0}: Missing SCM dependencies?", repository);
         }
     }
 
-    private void createHistoryCacheReal(Collection<Repository> repositories) {
+    private Map<Repository, Optional<Exception>> createHistoryCacheReal(Collection<Repository> repositories) {
```

Comment:
Can we add repositories with exception only to the map, instead of keeping the value as optional

Reply:
This way I like it better because it makes it more apparent what repositories were processed, also good for testing.